Add initial set of tests

kborchers opened the pull request:
Sets up Enzyme, Mocha, Chai tests run via Grunt. Also moves linting to a Grunt task.
Fixes #1

rxaviers commented:
Kris, great work on getting tests setup here! :tada: 

About `grunt`, given our tasks are quite simple, I'm wondering if we could drop using grunt and make use of npm script tasks instead. They are simpler to maintain. Obviously, this could be tackled as a separate issue if you don't want to touch it for now.

rxaviers commented on test/formatCurrency.spec.js:
```diff
@@ -0,0 +1,24 @@
+    require( "cldr-data/main/en/currencies" ),
+    require( "cldr-data/supplemental/currencyData" ),
+    require( "cldr-data/supplemental/likelySubtags" )
+);
```
I assume the specificity of CLDR content isn't subject to react-globalize tests goal, so I would simply replace these with:

```js
Globalize.load(
  require( "cldr-data" ).entireSupplemental(),
  require( "cldr-data" ).entireMainFor( "en" )
);
```

Ideally, we could place this in a `before` step for all tests and don't need to write this in every files. :)

rxaviers commented on test/formatDate.spec.js:
```diff
@@ -0,0 +1,26 @@
+
+    it("formats date as 1/1/2016", () => {
+        const wrapper = shallow(<FormatDate locale="en" pattern="GyMMMd">{new Date("Jan 01 2016")}</FormatDate>);
+        expect(wrapper.text()).to.eql("1/1/2016");
```
Let's use `equal` instead of `eql` (deeply equal to a value)?

kborchers commented:
> I'm wondering if we could drop using grunt and make use of npm script tasks instead

I would be fine with that but maybe create an issue and do that separately since it would impact more than just tests

rxaviers commented on test/formatMessage.spec.js:
```diff
@@ -0,0 +1,62 @@
+        const wrapper = shallow(<FormatMessage locale="en" path="task" variables={{count: 1}} />);
+        expect(wrapper.text()).to.eql("You have one task remaining");
+    });
+});
```
We need tests using default message (no `path` attribute).

kborchers replied:
This wasn't meant to be a comprehensive test suite but instead a start and we can add as we go. That said, I will add that test as it will fail during development of the solution for #34 which will be helpful

rxaviers commented on test/formatCurrency.spec.js:
```diff
@@ -0,0 +1,24 @@
+
+describe("formatCurrency Component", () => {
+    it("renders as a <span>", () => {
+        const wrapper = shallow(<FormatCurrency locale="en" currency="USD">{150}</FormatCurrency>);
```
Similarly, we can set the default locale as English with `Globalize.locale( "en" )` in a preliminary `before` step for all tests and get rid of this attribute in all tests.

Perhaps one test could assert the optional `locale` attribute to ensure it overrides the default one.

rxaviers commented on test/formatDate.spec.js:
```diff
@@ -0,0 +1,26 @@
+    });
+
+    it("formats date as 1/1/2016", () => {
+        const wrapper = shallow(<FormatDate locale="en" pattern="GyMMMd">{new Date("Jan 01 2016")}</FormatDate>);
```
Is this working? :frowning: I was expecting `options` instead of `pattern`.

kborchers replied:
Yep, definitely passes locally.

kborchers replied:
BTW, this was pulled directly from the example so if that is incorrect behavior, when fixed the example also needs to be updated.

rxaviers replied:
I believe you have Globalize 1.0 locally, let's get tests implemented for 1.1.x.

kborchers replied:
Completely removed `node_modules` and did a fresh `npm install`. Verified I am on Globalize 1.1.1 and all tests still pass.

rxaviers replied:
Above you're importing from `dist`, so have you re-built it please? By the way, I think we should test `src` instead.

kborchers replied:
It is rebuilt on every run so `dist` is up to date. I have changed locally to use `src` and all tests pass though I have to remove lines like this https://github.com/jquery-support/react-globalize/blob/master/src/currency.js#L3 which I am honestly not sure why they would be there and what exactly they would be importing. After removal though, all tests pass.

rxaviers replied:
The tested value doesn't correspond to the given pattern. The tested value is the default value when no options is passed. It would be really weird if this worked, I was starting to get concerned wondering that react-globalize had developed AI. :smile: :stuck_out_tongue_closed_eyes: See https://github.com/jquery-support/react-globalize/blob/master/src/date.js#L5

rxaviers replied:
About the lines you have removed, they are there because of esperanto (kind of a hack for esperanto to correctly build it for AMD), we'll need to investigate how to fix this.

rxaviers commented:
Are there any options that we list in our documentation (i.e., README.md) that are not being tested?

rxaviers commented:
> Are there any options that we list in our documentation (i.e., README.md) that are not being tested?

Ah! One is `elements` attributes for `FormatMessage`.

kborchers commented:
> Are there any options that we list in our documentation (i.e., README.md) that are not being tested?

Definitely. Like I mentioned in a previous line comment, this was not meant to be a comprehensive set of tests on this first pass. My goal was to get the structure in place and some initial tests and folks can add tests as they have time.

rxaviers commented on package.json:
```diff
@@ -36,12 +34,24 @@
     "eslint": "^1.10.3",
     "eslint-config-defaults": "^7.1.1",
     "eslint-plugin-react": "^3.11.2",
+    "globalize": ">= 1.0.0 || 1.1.0-alpha - 1.1.x",
```
Let's stick with `1.1.x` here in devDependencies.

rxaviers commented:
Great job so far!

kborchers commented:
I believe I have addressed all comments except the following:
* still need more tests but we can add as we go
* did not pull from `src` as we need to figure out how to properly address the esperanto/AMD issues

Otherwise, should be good to go and squashed the commits as well

rxaviers commented:
@kborchers, it LGTM :+1:.
